src/sentry/static/sentry/app/views/organizationDashboard/discoverQuery.jsx

```diff
@@ -68,25 +87,45 @@ class DiscoverQuery extends React.Component {
 
   componentDidUpdate(prevProps) {
     const keysToIgnore = ['children'];
+
+    // Ignore "busy" and "releases" props if we are not waiting for releases
+    // Otherwise we can potentially make an extra request if busy !== nextBusy and
+    // globalSelection === nextGlobalSelection
+    if (!this.doesRequireReleases(this.props.queries)) {
+      keysToIgnore.push('busy');
+      keysToIgnore.push('releases');
+    }
+
     if (isEqual(omit(prevProps, keysToIgnore), omit(this.props, keysToIgnore))) {
       return;
     }
 
-    if (this.props.releases !== prevProps.releases) {
-      this.createQueryBuilders();
-    } else {
-      this.fetchData();
-    }
+    this.createQueryBuilders();
+    this.fetchData();
   }
 
   componentWillUnmount() {
     this.queryBuilders.forEach(builder => builder.cancelRequests());
+    // Cleanup query builders
+    this.queryBuilders = [];
   }
 
+  // Checks queries for any that are dependent on recent releases
+  doesRequireReleases = memoize(
+    queries =>
+      !!queries.find(
+        ({constraints}) => constraints && constraints.includes('recentReleases')
+      )
+  );
+
   createQueryBuilders() {
     const {organization, queries} = this.props;
+
+    this.queryBuilders = [];
+
     queries.forEach(({constraints, ...query}) => {
       if (constraints && constraints.includes('recentReleases')) {
+        // Can't create query yet because no releases
```

Comment:
This feels like a class of `busy` status situations - should the parent be passing busy=true rather that relying on this component to check here whether there are release constraints and the releases are not loaded?

Reply:
I think the querying logic should be held in this component, its parents don't need to know details about the queries itself.

We could change the name of the `busy` prop to something more specific to releases state if that would help clarify things.